src/main.c

```diff
@@ -2042,6 +2112,43 @@ static lbm_value ext_set_fw_version(lbm_value *args, lbm_uint argn) {
     return VESC_IF->lbm_enc_sym_true;
 }
 
+// Called from Lisp to pass in values from the bms. If no prameters are called, return
+// tiltback_bms_enabled.
+static lbm_value ext_bms(lbm_value *args, lbm_uint argn) {
+    Data *d = (Data *) ARG;
+    d->bms_fault = BMSF_NONE;
```

Comment:
The way you update `d->bms_fault` by resetting it here and later setting the error flags again is not atomic and will lead to unnecessary races. `d->bms_fault` should be updated in a single operation to make it atomic on the stm32. Otherwise, if there is e.g. some timing hooked onto some of the faults, like a timeout on the HV pushback, it could get accidentally reset if a loop iteration runs between the above reset and the set of the fault that follows below.